src/servers/http_server.cc

```diff
@@ -801,6 +801,24 @@ ValidateOutputParameter(triton::common::TritonJson::Value& io)
   return nullptr;  // success
 }
 
+TRITONSERVER_Error*
+CheckInputElementCount(
+    const std::vector<int64_t>& shape,
+    triton::common::TritonJson::Value& tensor_data)
+{
+  int shape_product = 1;
+  for (size_t i = 0; i < shape.size(); i++) {
+    shape_product *= shape[i];
+  }
+  if (shape_product != static_cast<int>(tensor_data.ArraySize())) {
```

Comment:
Convert it to `int` may loss precision?

Reply:
Note that [TensorData](https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md#tensor-data) can be represent in two ways, so the function should check which representation it uses and compare the dimensions accordingly:

- if 1-D, then get element count of the shape and compare the value
- if > 1-D, then expect the number of dimensions is the same and iterate each dimension to ensure they have the same shape.

Reply:
Yes. Good catch @GuanLuo!